src/library/dao/cratedao.cpp

```diff
@@ -18,6 +18,32 @@ CrateDAO::~CrateDAO() {
 
 void CrateDAO::initialize() {
     qDebug() << "CrateDAO::initialize()";
+
+    //get the count to allocate HashMap
+    int tracksInCratesCount = 0;
+    QSqlQuery query(m_database);
+    query.prepare("SELECT COUNT(*) from " CRATE_TRACKS_TABLE);
+
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query);
+    }
+
+    tracksInCratesCount = query.value(0).toInt();
+
+    m_cratesTrackIsIn.reserve(tracksInCratesCount);
+
+    //now fetch all Tracks from all crates and insert them into the hashmap
+    query.prepare("SELECT track_id, crate_id from " CRATE_TRACKS_TABLE);
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query);
+    }
+
+    const int trackIdColumn = query.record().indexOf("track_id");
+    const int crateIdColumn = query.record().indexOf("crate_id");
+    while (query.next()) {
+        m_cratesTrackIsIn.insert(query.value(crateIdColumn).toInt(),
+                                 query.value(trackIdColumn).toInt());
+    }
```

Comment:
> This is finally a decision of taste. If you have a working solution that has a reasonable CPU load it is a good one. 

At the moment this branch contains a working solution. I've worked with it a while without performance problems. I'm on an i5 with 16Gig RAM so your atom would be the better test-case. 

> Take the on that suits more to you.
As the in-memory table is ready now and tested for me, I would stick with that one for now. Once we have Nazars work in it would be worth refactoring that though - having multiple datasets that have to be in sync is clearly a point of failure.

So what about a test on your atom platform and when scolling by holding down arrowkeys or sth. like this isn't slowed down merge this as is and else test the other solution to see whats better?

I've just looked in nazars branch... will take a while to get into this...